### DISCUSSION TITLE
Allow for Delta Runs

### DISCUSSION THREAD

**Discussion by JeroenKnoops** [mesh-iit/outside-collaborators#175] [Ideas]
## Problem

We're making a great use of this template.
We're very happy, but in a large organization, this solution does not scale.

When adding 1 person to a group, all repositories are being checked and the run can take up to 3 hours.

## Solution

Instead of running everything for all repositories and users, we can also only run the scripts on the files which are changed in the PR.

### How to determine which files are necessary

We've created a CLI tool to determine which files need to be included for certain changes.
When a `repos` file has been changed, we're looking in the `groups` folder for files which are used for those `repos`. Same for changes in a `groups` file, we're looking into the `repos` folder for relevant files until we found all relevant files.

### Process
We're removing all files in `repos` and `groups` and only add the found files and process as normal.

Every night we run a complete check.
We can trigger a complete run of x hours manually, if needed.

### Used tools

- Action to see which files are changed in a PR: [`tj-actions/changed-files@v35`](https://github.com/tj-actions/changed-files)
- CLI tool (written in Rust) to find the files
- Slack Notificator when complete files are removed, manual actions are required
- Unmodified Outside-collaborators scripts

## Share

Do you want me to share this code in a separate repository so people can use these delta runs as well?

**Comment by pattacini** [comment]
Fantastic idea @JeroenKnoops 🚀 
I'd really like to include your contribution!

>Do you want me to share this code in a separate repository so people can use these delta runs as well?

Yep, please, go on, and once ready proceed with a PR.
I can help, of course.
